core/src/com/unciv/logic/automation/ConstructionAutomation.kt

```diff
@@ -168,12 +168,9 @@ class ConstructionAutomation(val cityConstructions: CityConstructions){
                 && it.isBuildable(cityConstructions)
                 && Automation.allowSpendingResource(civInfo, it) }
         if (workerEquivalents.isEmpty()) return // for mods with no worker units
-        if (civInfo.getIdleUnits().any { it.isAutomated() && it.hasUniqueToBuildImprovements })
-            return // If we have automated workers who have no work to do then it's silly to construct new workers.
```

Comment:
Huh. Neat. So I guess it was possible to flat-out disable an AI's improvement construction by trapping a worker in a bubble inside your borders and killing the other two. (Not a review comment, but I find the exploit interesting.)